front/lib/api/assistant/agent.ts

```diff
@@ -42,15 +102,188 @@ export async function createAgentConfiguration(
     action?: AgentActionConfigurationType;
```

Comment:
As per IRL

should be 

createAgentGenerationConfiguration
createAgentActionConfiguration

and createAgentConfiguration that takes the full types that will have their `ids` set.